### PR TITLE
Mysql faster

### PR DESCRIPTION
**Pull request by jakob-r**
* import-full-data.sh progress bar if pv is installed
* Docker directly inherits from ubuntu
* use data.table (instead of data.frame) internally
* removed some helper functions in data_acess.R for better readability
* made numerical algo_id (implementation.id) in database obsolete
* improved SQL Query for get_parameter_table: Does everything in one command
* memoise caching without time limit but on disk
* renaming: algo_id is always something like "mlr.classif.ranger"
* renaming: parameter etc. to par_vals if they are actual values to not get confused with parameters that are the definition

### CODE REVIEW

**Comment by jakob-r** [comment]
@mboecker Please look at everything. If you totally dislike something change it / tell me.

File: docker/omlbotlookup/data_access.R [hunk 1]
```diff
@@ -23,22 +23,23 @@ con <- dbConnect(MySQL(), user = mysql_username, password = mysql_password, dbna
 # See /docker/preparation/prepare_parameter_ranges.R for instructions.
 # This file contains parameter range data obtained from the omlbot-sourcecode.
 parameter_ranges = readRDS("parameter_ranges.Rds")
+names(parameter_ranges) = paste0("mlr.", names(parameter_ranges))
```
Comment (mboecker) [inline docker/omlbotlookup/data_access.R]: Das würde ich lieber direkt in der "parameter_ranges" machen.

File: docker/omlbotlookup/data_access.R [hunk 2]
```diff
@@ -23,22 +23,23 @@ con <- dbConnect(MySQL(), user = mysql_username, password = mysql_password, dbna
...
 
 
 #' Return a list of parameter definitions. This list contains every necessary parameter for the given algorithm.
 #'
 #' @param task_id This is `task_id` from the table `run` in the database.
 #'
 #' @return A named list with one entry for each needed parameter.
-get_params_for_algo = function(algo_name) {
+get_params_for_algo = function(algo_id) {
```
Comment (mboecker) [inline docker/omlbotlookup/data_access.R]: algo_id ist eigentlich ungleich algo_name. Ein algo_name entspricht ja mehreren algo_ids.

Comment (mboecker) [reply docker/omlbotlookup/data_access.R]: Ok du willst also eigentlich die option entfernen, nach einzelnen algo_ids zu fragen. find ich in ordnung, allerdings hätte ich da den namen algo_name bevorzugt. ist aber ja im endeffekt egal.

**Comment by mboecker** [comment]
By the way, the time for the first request is substantially higher for me now, but I think this is irrelevant if the queries afterwars are faster?

**Pull request closed without merging**